### PR TITLE
build: pass through triggers

### PR DESCRIPTION
**Pull request by jazzdan**
~not ready for review yet~

Good to go!

### CODE REVIEW

File: internal/build/boil_steps.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package build
+
+import (
+	"github.com/windmilleng/tilt/internal/model"
+)
+
+type BoiledStep struct {
+	cmd         model.Cmd
+	pathMapping pathMapping
```
Comment (nicks) [inline internal/build/boil_steps.go]: i don't understand what the goal of this pathMapping property is? afaict, it's never used

File: internal/build/boil_steps.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+func BoilSteps(steps []model.Step, pathMappings []pathMapping) ([]BoiledStep, error) {
+	res := []BoiledStep{}
+	for _, step := range steps {
+		if step.Trigger == nil {
+			res = append(res, BoiledStep{cmd: step.Cmd})
+			continue
+		}
+		for _, pm := range pathMappings {
+			matches, err := step.Trigger.Matches(pm.LocalPath, false)
```
Comment (nicks) [inline internal/build/boil_steps.go]: it seems like you could pass in the LocalPath as FilesChanged(), like you had before? you don't use the rest of the pathMapping

File: internal/build/boil_steps.go (hunk 1 above)
Comment (jazzdan) [reply internal/build/boil_steps.go]: Yeah looks like we don't need this anymore, I'll remove it.

File: internal/build/container_updater.go [hunk 3]
```diff
@@ -60,6 +60,7 @@ func (r *ContainerUpdater) UpdateInContainer(ctx context.Context, cID k8s.Contai
 
 	// Exec steps on container
 	for _, s := range steps {
+		fmt.Printf("Executing step %v in container\n", s.Argv)
```
Comment (nicks) [inline internal/build/container_updater.go]: :eyes:

File: internal/engine/local_container_build_and_deployer.go [hunk 4]
```diff
@@ -59,7 +59,13 @@ func (cbd *LocalContainerBuildAndDeployer) BuildAndDeploy(ctx context.Context, s
 		return BuildResult{}, err
 	}
 	logger.Get(ctx).Infof("  → Updating container…")
-	err = cbd.cu.UpdateInContainer(ctx, cID, cf, model.BoilStepsTODO(service.Steps))
+	boiledSteps, err := build.BoilSteps(service.Steps, cf)
+	if err != nil {
+		return BuildResult{}, err
+	}
+
+	fmt.Printf("Boiled steps: %+v \n", boiledSteps)
```
Comment (nicks) [inline internal/engine/local_container_build_and_deployer.go]: :eyes:

**Pull request merged**